src/compiler/utilities.ts

```diff
@@ -2591,6 +2437,464 @@ namespace ts {
+            || isExpressionKind(kind);
+    }
+
+    export function isForInitializer(node: Node): node is ForInitializer {
+        return isForInitializerKind(node.kind);
+    }
+
+    export function isCaseBlock(node: Node): node is CaseBlock {
+        return node.kind === SyntaxKind.CaseBlock;
+    }
+
+    export function isBlock(node: Node): node is Block {
+        return node.kind === SyntaxKind.Block;
+    }
+
+    export function isCatchClause(node: Node): node is CatchClause {
+        return node.kind === SyntaxKind.CatchClause;
+    }
+
+    export function isVariableDeclaration(node: Node): node is VariableDeclaration {
+        return node.kind === SyntaxKind.VariableDeclaration;
+    }
+
+    export function isEnumMember(node: Node): node is EnumMember {
+        return node.kind === SyntaxKind.EnumMember;
+    }
+
+    function isModuleNameKind(kind: SyntaxKind) {
+        return kind === SyntaxKind.Identifier
+            || kind === SyntaxKind.StringLiteral;
+    }
+
+    export function isModuleName(node: Node): node is ModuleName {
+        return isModuleNameKind(node.kind);
+    }
+
+    function isCaseOrDefaultClauseKind(kind: SyntaxKind) {
+        return kind === SyntaxKind.CaseClause
+            || kind === SyntaxKind.DefaultClause;
+    }
+
+    export function isCaseOrDefaultClause(node: Node): node is CaseOrDefaultClause {
+        return isCaseOrDefaultClauseKind(node.kind);
+    }
+
+    function isModuleReferenceKind(kind: SyntaxKind) {
+        return kind === SyntaxKind.ExternalModuleReference
+            || isEntityNameKind(kind);
+    }
+
+    export function isModuleReference(node: Node): node is ModuleReference {
+        return isModuleReferenceKind(node.kind);
+    }
+
+    export function isImportClause(node: Node): node is ImportClause {
+        return node.kind === SyntaxKind.ImportClause;
+    }
+
+    function isNamedImportBindingsKind(kind: SyntaxKind) {
+        return kind === SyntaxKind.NamedImports
+            || kind === SyntaxKind.NamespaceImport;
+    }
+
+    export function isNamedImportBindings(node: Node): node is NamedImportBindings {
+        return isNamedImportBindingsKind(node.kind);
+    }
+
+    export function isImportSpecifier(node: Node): node is ImportSpecifier {
+        return node.kind === SyntaxKind.ImportSpecifier;
+    }
+
+    export function isNamedExports(node: Node): node is NamedExports {
+        return node.kind === SyntaxKind.NamedExports;
+    }
+
+    export function isExportSpecifier(node: Node): node is ExportSpecifier {
+        return node.kind === SyntaxKind.ExportSpecifier;
+    }
+
+    export function isJsxOpeningElement(node: Node): node is JsxOpeningElement {
+        return node.kind === SyntaxKind.JsxOpeningElement;
+    }
+
+    export function isJsxClosingElement(node: Node): node is JsxClosingElement {
+        return node.kind === SyntaxKind.JsxClosingElement;
+    }
+
+    function isJsxChildKind(kind: SyntaxKind) {

```

Comment:
Not offhand, just got into the habit of splitting out the checks. Initially I split out the checks simply against `kind` to avoid the cost of looking up `node.kind`.